File: chain-sdk/clojure/src/on_chain_api/specs/uniswap_v3_swap_v3_dto.clj
(ns on-chain-api.specs.uniswap-v3-swap-v3-dto
  (:require [clojure.spec.alpha :as s]
            [spec-tools.data-spec :as ds]
            [on-chain-api.specs.numerics-big-integer :refer :all]
            [on-chain-api.specs.numerics-big-integer :refer :all]
            [on-chain-api.specs.numerics-big-integer :refer :all]
            [on-chain-api.specs.transactions-e-trade-aggressive-side :refer :all]
            )
  (:import (java.io File)))


(def uniswap-v3-swap-v3-dto-data
  {
   (ds/opt :entry_time) inst?
   (ds/opt :recv_time) inst?
   (ds/opt :block_number) int?
   (ds/opt :vid) int?
   (ds/opt :id) string?
   (ds/opt :transaction) string?
   (ds/opt :timestamp) inst?
   (ds/opt :pool) string?
   (ds/opt :token_0) string?
   (ds/opt :token_1) string?
   (ds/opt :sender) string?
   (ds/opt :recipient) string?
   (ds/opt :origin) string?
   (ds/opt :amount_0) string?
   (ds/opt :amount_1) string?
   (ds/opt :amount_usd) string?
   (ds/opt :sqrt_price_x96) numerics-big-integer-spec
   (ds/opt :tick) numerics-big-integer-spec
   (ds/opt :log_index) numerics-big-integer-spec
   (ds/opt :evaluated_price) float?
   (ds/opt :evaluated_amount) float?
   (ds/opt :evaluated_aggressor) transactions-e-trade-aggressive-side-spec
   (ds/opt :pool_id) string?
   (ds/opt :transaction_id) string?
   })

(def uniswap-v3-swap-v3-dto-spec
  (ds/spec
    {:name ::uniswap-v3-swap-v3-dto
     :spec uniswap-v3-swap-v3-dto-data}))

File: chain-sdk/javascript-closure-angular/API/Client/UniswapV3SwapV3DTO.js
goog.provide('API.Client.UniswapV3SwapV3DTO');

/**
 * Swap are created for each token swap within a pair.
 * @record
 */
API.Client.UniswapV3SwapV3DTO = function() {}

/**
 * @type {!Date}
 * @export
 */
API.Client.UniswapV3SwapV3DTO.prototype.entryTime;

/**
 * @type {!Date}
 * @export
 */
API.Client.UniswapV3SwapV3DTO.prototype.recvTime;

/**
 * Number of block in which entity was recorded.
 * @type {!number}
 * @export
 */
API.Client.UniswapV3SwapV3DTO.prototype.blockNumber;

/**
 * 
 * @type {!number}
 * @export
 */
API.Client.UniswapV3SwapV3DTO.prototype.vid;

/**
 * Identifier, format: transaction hash + \"#\" + index in swaps Transaction array.
 * @type {!string}
 * @export
 */
API.Client.UniswapV3SwapV3DTO.prototype.id;

/**
 * Pointer to transaction.
 * @type {!string}
 * @export
 */
API.Client.UniswapV3SwapV3DTO.prototype.transaction;

/**
 * Timestamp of transaction.
 * @type {!Date}
 * @export
 */
API.Client.UniswapV3SwapV3DTO.prototype.timestamp;

/**
 * Pool swap occured within.
 * @type {!string}
 * @export
 */
API.Client.UniswapV3SwapV3DTO.prototype.pool;

/**
 * Reference to token0 as stored in pair contract.
 * @type {!string}
 * @export
 */
API.Client.UniswapV3SwapV3DTO.prototype.token0;

/**
 * Reference to token1 as stored in pair contract.
 * @type {!string}
 * @export
 */
API.Client.UniswapV3SwapV3DTO.prototype.token1;

/**
 * Sender of the swap.
 * @type {!string}
 * @export
 */
API.Client.UniswapV3SwapV3DTO.prototype.sender;

/**
 * Recipient of the swap.
 * @type {!string}
 * @export
 */
API.Client.UniswapV3SwapV3DTO.prototype.recipient;

/**
 * Transaction origin: the EOA (Externally Owned Account) that initiated the transaction
 * @type {!string}
 * @export
 */
API.Client.UniswapV3SwapV3DTO.prototype.origin;

/**
 * Delta of token0 swapped.
 * @type {!string}
 * @export
 */
API.Client.UniswapV3SwapV3DTO.prototype.amount0;

/**
 * Delta of token1 swapped.
 * @type {!string}
 * @export
 */
API.Client.UniswapV3SwapV3DTO.prototype.amount1;

/**
 * Derived amount of tokens sold in USD.
 * @type {!string}
 * @export
 */
API.Client.UniswapV3SwapV3DTO.prototype.amountUsd;

/**
 * @type {!API.Client.NumericsBigInteger}
 * @export
 */
API.Client.UniswapV3SwapV3DTO.prototype.sqrtPriceX96;

/**
 * @type {!API.Client.NumericsBigInteger}
 * @export
 */
API.Client.UniswapV3SwapV3DTO.prototype.tick;

/**
 * @type {!API.Client.NumericsBigInteger}
 * @export
 */
API.Client.UniswapV3SwapV3DTO.prototype.logIndex;

/**
 * @type {!number}
 * @export
 */
API.Client.UniswapV3SwapV3DTO.prototype.evaluatedPrice;

/**
 * @type {!number}
 * @export
 */
API.Client.UniswapV3SwapV3DTO.prototype.evaluatedAmount;

/**
 * @type {!API.Client.TransactionsETradeAggressiveSide}
 * @export
 */
API.Client.UniswapV3SwapV3DTO.prototype.evaluatedAggressor;

/**
 * @type {!string}
 * @export
 */
API.Client.UniswapV3SwapV3DTO.prototype.poolId;

/**
 * @type {!string}
 * @export
 */
API.Client.UniswapV3SwapV3DTO.prototype.transactionId;

